README.md

```diff
@@ -212,17 +212,21 @@ type Authorization struct {
   s       VarSig
 }
 
-type Invocation struct {
+type Context struct {
   v       SemVer
   run     &Task
   meta    {String : Any}
   prf     [&UCAN]
-  auth    &Authorization
 
   # Receipt of the invocation that caused this invocation
   cause   optional &Invocation
 }
 
+type Invocation struct {
+  ctx     Context
+  auth    &Authorization
+}
+
```

Comment:
Yeah, I'm fairly convinced that this is a requirement regardless of how we want to write worflows. The concern with the existing system is that you need to sign all of the sub-invoctaions because the resource limits are at the same level as the signature. This changes the CID, so you can't sign the Workflow wrapper at the same time without finding a hash cycle.

By pulling the auth to another layer, we get the flexibility to do nesting with a single signature without breaking the signature reference mechanism as it exists in `v0.1`.

I actually don't mind if the signature is subtyped or wrapped. I just need a struct that's everything minus the signature, and IPFS Schema doesn't have subtying/splats:

```haskell
type Context struct {
  v    Semver
  run  &Task
  meta {String : Any}
  prf  [&UCAN]
}

-- Subtyped

type Invocation struct { -- Exactly the same as today
  ...Context
  auth &Authorization
}

-- Wrapped

type Invocation struct {
  ctx  Context -- could also be a Link, but this is closer to today
  auth &Authorization
}
```

The added bonus is that this also signs over all fields, and is really really clear about that fact